pkg/orchestration/wiring/wiringutil/compute/secret_test.go

```diff
@@ -0,0 +1,223 @@
+		Resource: "my-compute--my-db--binding",
+		Path:     "data.username",
+		Modifier: smith_v1.ReferenceModifierBindSecret,
+	}
+	expectedEnvVars := map[string]string{
+		"SQS_QUEUE1_A_B_C":          fmt.Sprintf("!{my-compute--my-queue--binding-%s}", makeRefPathSuffix("data.abc[0]")),
+		"SQS_MY_QUEUE_A_B_C_2":      fmt.Sprintf("!{my-compute--my-queue--binding-%s}", makeRefPathSuffix("data.abc[1]")),
+		"SQS_MY_QUEUE_ABC":          fmt.Sprintf("!{my-compute--my-queue--binding-%s}", makeRefPathSuffix("data.abc[2]")),
+		"SQS_MY_QUEUE_MY_OTHER_VAR": fmt.Sprintf("!{my-compute--my-queue--binding-%s}", makeRefPathSuffix("data.myOtherVar")),
+		"PG_MY_DB_USERNAME":         fmt.Sprintf("!{my-compute--my-db--binding-%s}", makeRefPathSuffix("data.username")),
+	}
+
+	// do
+	smithReferences, envVars, err := GenerateEnvVars(renameMap, []BindingResult{queueBinding, databaseBinding})
+	require.NoError(t, err)
+
+	// results
+	require.Len(t, smithReferences, 5)
+	assert.Contains(t, smithReferences, expectedReference1)
+	assert.Contains(t, smithReferences, expectedReference2)
+	assert.Contains(t, smithReferences, expectedReference3)
+	assert.Contains(t, smithReferences, expectedReference4)
+	assert.Contains(t, smithReferences, expectedReference5)
+	assert.Equal(t, expectedEnvVars, envVars)
+}
+
+func TestGenerateEnvVarsEmptyBindings(t *testing.T) {
+	t.Parallel()
+
+	// Does nothing if empty - this implies there were no dependencies, thus there are no environment variables
+	smithReferences, envVars, err := GenerateEnvVars(map[string]string{}, []BindingResult{})
+	require.NoError(t, err)
+	assert.Len(t, smithReferences, 0)
+	assert.Len(t, envVars, 0)
+}
+
+func TestGenerateEnvVarsEmptyVars(t *testing.T) {
+	t.Parallel()
+
+	// given
+	queueBinding := constructBindingResult("my-compute", "my-queue", "my-queue--instance", "SQS", map[string]string{})
+
+	// do
+	smithReferences, envVars, err := GenerateEnvVars(map[string]string{}, []BindingResult{queueBinding})
+	require.NoError(t, err)
+
+	// results
+	assert.Len(t, smithReferences, 0)
+	assert.Len(t, envVars, 0)
+}
+
+func constructBindingResult(consumerName string, dependencyName string, smithServiceInstanceName string, prefix string, vars map[string]string) BindingResult {
+	return BindingResult{
+		// This binding result describes a dependency to a resource in the State
+		ResourceName: voyager.ResourceName(dependencyName),
+
+		// The shape is the shape output by that particular resource
+		// In this case, it's always a BindableEnvVar shape.
+		BindableEnvVarShape: &knownshapes.BindableEnvironmentVariables{
+			ShapeMeta: wiringplugin.ShapeMeta{
+				ShapeName: knownshapes.BindableEnvironmentVariablesShape,
+			},
+			Data: knownshapes.BindableEnvironmentVariablesData{
+				// This describes the instance that we should bind to
+				BindableShapeStruct: wiringplugin.BindableShapeStruct{
+					ServiceInstanceName: wiringplugin.ProtoReference{
+						Resource: smith_v1.ResourceName(smithServiceInstanceName),
```

Comment:
This is not right. Instance name and resource name may differ.

Reply:
Huh? This is the ProtoReference. The resource that the reference is referring to for the binding is the service instance name.

Reply:
This is a confusing test =) I fixed the types of the arguments and renamed them and everything still works. `smithServiceInstanceName` looked to me like the object metadata name of a ServiceInstance object, not a Smith resource name.